src/Datasource/PaginatedResultSet.php

```diff
@@ -0,0 +1,143 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Datasource;
```

Comment:
Should this class be in the `Paging` namespace with other paging related classes? Or should the new `Paging` namespace be `Datasource\Paging`?

Reply:
The discontinuity is a little confusing to me as well. I can see maybe Paging was meant as an interface-only, but I'm not sure we need that kind of separation from the result set. I fear we're going to get into more Interface + Trait combinations to try and satisfy these split dependencies.

Reply:
Initially I had thought of having `Paging` as a separate package. That would mean adding it as a dependency of the `Datasource` package.

But if the consensus is that having a separate paging package isn't necessary or will be useful enough then I can simplify things and move `Paging` to `Datasource/Paging` instead.